test/test_context.py

```diff
@@ -60,37 +62,37 @@ def test_context1():
     check_evaluation("bar[]", "42", to_string_expr=False, to_string_expected=False)
 
 
-def test_context2():
-    nomessage = tuple([])
-    for expr, expected, lst_messages, msg in [
+@pytest.mark.parametrize(
```

Comment:
I reformulated this test as a parametrized test because it was easier to debug...